src/lib/journey/journey.store.ts

```diff
@@ -132,9 +137,21 @@ export function initialize(initOptions?: StepOptions): JourneyStore {
     }
 
     if (nextStep.type === StepType.Step) {
-      /**
-       * SUCCESSFUL CONTINUATION BLOCK
-       */
+      const stageAttribute = nextStep.getStage();
+
+      let stageJson: Maybe<Record<string, unknown>> = null;
+
+      // Check if stage attribute is serialized JSON
+      if (stageAttribute && stageAttribute.includes('{')) {
+        try {
+          stageJson = JSON.parse(stageAttribute);
+        } catch (err) {
+          console.warn('Stage attribute value was not parsable');
+        }
+      }
+
+      const callbackMetadata = buildCallbackMetadata(nextStep, initCheckValidation(), stageJson);
+      const stepMetadata = buildStepMetadata(callbackMetadata, stageJson);
```

Comment:
This is the logic for extracting the data out of a stage attribute and passing it to the metadata functions.